app/assets/javascripts/discourse/app/components/choose-message.gjs

```diff
@@ -49,32 +43,40 @@ export default class ChooseMessage extends Component {
       </label>
 
       <input
-        {{on "input" this.search}}
+        {{on "input" this.setSearchTerm}}
         type="text"
         placeholder={{i18n "choose_message.title.placeholder"}}
         id="choose-message-title"
       />
 
-      {{#if this.loading}}
-        <p>{{i18n "loading"}}</p>
-      {{else if this.hasSearched}}
-        {{#each this.messages as |message|}}
-          <div class="controls existing-message">
-            <label class="radio">
-              <input
-                {{on "click" (fn @setSelectedTopicId message)}}
-                type="radio"
-                name="choose_message_id"
-              />
-              <span class="message-title">
-                {{message.title}}
-              </span>
-            </label>
-          </div>
-        {{else}}
-          <p>{{i18n "choose_message.none_found"}}</p>
-        {{/each}}
-      {{/if}}
+      <AsyncContent
+        @asyncData={{this.search}}
+        @context={{this.searchedTitle}}
```

Comment:
How about using `fn` for this? Then, `AsyncContent` can be much simpler: it just calls the function its given, with no additional arguments.

```suggestion
        @asyncData={{fn this.search this.searchedTitle}}
```

Reply:
I tried, but updating `searchedTitle` does not trigger a new search.

`fn` is somehow making it non-reactive